server/src/test/java/org/spine3/server/storage/AbstractStorageShould.java

```diff
@@ -152,6 +165,42 @@ public void rewrite_record_if_write_by_the_same_id() {
         writeAndReadRecordTest(id);
     }
 
+    @Test
+    public void have_index_on_ID() {
+        final Iterator<I> index = storage.index();
+        assertNotNull(index);
+    }
+
+    @Test
+    public void index_all_IDs() {
+        final int recordCount = 10;
+        final Set<I> ids = new HashSet<>(recordCount);
+        for (int i = 0; i < recordCount; i++) {
+            final I id = newId();
+            writeRecord(id);
+            ids.add(id);
+        }
+
+        final Iterator<I> index = storage.index();
+        final Collection indexValues = Sets.newHashSet(index);
+
+        assertEquals(ids.size(), indexValues.size());
+        Verify.assertContainsAll(indexValues, ids.toArray());
+    }
+
+    @Test
```

Comment:
In this case, we rely on a Guava Collection implementation of `Iterator`. In one particular case  (in `InMemoryEventStorage`) we use a collection which throws `IllegalStateException`.
This is the reason why we don't use `expected` - it does not support multiple `Exception` types.
I don't like this either.